ucum/ucum.go

```diff
@@ -0,0 +1,353 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ucum provides UCUM (Unified Code for Units of Measure) support for the CQL engine.
+package ucum
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// CQLToUCUMDateUnits maps CQL date units to their UCUM equivalents
+var CQLToUCUMDateUnits = map[string]string{
+	"years":        "a_g",
```

Comment:
I think this is right, but i cant remember / find the right docs on some of the specifics here. Can you add comments on why the `_g` is added?

https://cql.hl7.org/02-authorsguide.html#quantities seems to suggest that "a" is the equivalent here. But I feel like I've also seen this syntax before.

Reply:
https://github.com/LHNCBC/ucum-lhc/blob/master/data/ucum.csv#L278

This is the npm package used in the javascript cql engine, which is where I started. Since go does not have a ucum package this file was intended to be a replacement for golang community not having the equivaleent package support.

Reply:
Thanks for the link here! Mind adding that link to the docs at the top of the file? It might help if we find further edge cases in the future. If this is what the Javascript CQL engine is using than I'm more apt to say we can go with this solution for now :)